### PR TITLE
add members to orgs via guild.xyz

### PR DESCRIPTION
**Pull request by CryptoGraffe**
### <samp>🤖 Generated by Copilot at 802b8b6</samp>

Added guild-based organization management and joining features. This involved adding `guild_id` and `guild_role_id` columns to the `public_organizations` table and the corresponding fields to the GraphQL schemas and queries. It also involved creating a `GuildSelector` component and updating various pages and components to use it. Additionally, it involved refactoring the `useOrgInviteToken` hook to `useOrgShareToken` and supporting both invite and welcome tokens for organizations.

### <samp>🤖 Generated by Copilot at 802b8b6</samp>

> _We're adding guilds to organizations, me hearties_
> _To make them more secure and fun_
> _We're changing hooks and types and migrations_
> _And we'll be done when the work is done_

## Test and Deployment Plan

Connect a guild to an org on the org settings page. Go to the members page (maybe have to refresh due to invalidation bug), then copy that welcome link and try to join it with another wallet. Try matching the guild criteria and also not matching it. 

### <samp>🤖 Generated by Copilot at 802b8b6</samp>

*  Added and updated migrations, metadata, and schema files for the `guild_id` and `guild_role_id` columns in the `organizations` table, which are used to store the guild association and role requirement for an organization. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-59b15e8a097faa04dfec696707c0c1720c6b18506e3e272b1c990647f0e4fb70R43-R44),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-59b15e8a097faa04dfec696707c0c1720c6b18506e3e272b1c990647f0e4fb70R74-R75),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-da1585335cae6870bb9d640b94afd59ac550de87efeb2d3eea4cb74af88b27e8R1-R4),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8c66122597b8b53297d07134b10a062dfa8c0835df27b49a17a2f387c5fb2d1dR1-R2),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-3959d62d5b478bee4be4ff097899a75fca321406f45af3d2a63fd081f96b1864R1-R4),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-247335793b0faefa0671f85b034f329b6f12d7b9e3c1c1091697eeadbfda41a2R1-R2),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20618-R20619),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20788-R20789),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20804-R20805),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20835-R20836),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20847-R20848),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20871-R20872),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20886-R20887),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20938-R20939),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R20976-R20985),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21025-R21026),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21045-R21046),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21055-R21056),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21065-R21066),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21091-R21092),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21111-R21112),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21133-R21142),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21198-R21199),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21208-R21209),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-85e7c26aa7c1055bc40e9fd67dfd49ad95964c817816aad24829f3bb1e8ec5a8R21218-R21219),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR8138-R8142),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR8158-R8162),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9259-R9260),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9348-R9349),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9362-R9369),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9391-R9392),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9428-R9437),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9470-R9471),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9497-R9498),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-8435c41f5069c32f84a110adf7ca73963708315c24f4aabdb0285e87471676ebR9513-R9517))
*  Added and updated the `handler` and `handleOrg` functions in the `createUserWithToken` action handler to include the `hasuraAddress` in the session variables and to check the user's membership in a guild if the token is a welcome link and the organization has a guild associated with it. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-facdccf70ca95c8fd27abae06840850cc36d240dec8aeab36d3607bf89d79018L37-R37),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-facdccf70ca95c8fd27abae06840850cc36d240dec8aeab36d3607bf89d79018L57-R57),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-facdccf70ca95c8fd27abae06840850cc36d240dec8aeab36d3607bf89d79018R122),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-facdccf70ca95c8fd27abae06840850cc36d240dec8aeab36d3607bf89d79018L129-R134),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-facdccf70ca95c8fd27abae06840850cc36d240dec8aeab36d3607bf89d79018R148-R161))
*  Added the `GuildSelector` component to display and handle the guild selection and role requirement for a circle or an organization in the settings pages. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-57b0e0735d32367af57ff74dbce332995ba2fad142b6c75b76f26fe700cda0a9R1-R139))
*  Updated the `OrgSettingsPage` component to use the `Form` and `ContentHeader` components as the outer wrapper and the sticky header, and to use the `GuildSelector` component to display and handle the guild association and role requirement for an organization. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL10-R20),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fR29),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fR55-R56),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL57-R74),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL78-R104),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fR166-R168),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fR178-R183),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL159-R215),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL291-R353))
*  Updated the `CircleAdminPage` component to use the `settings` variant of the `Panel` component to apply the panel styles, and to use the `GuildSelector` component to display and handle the guild association and role requirement for a circle. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL18-R19),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL58-L65),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL167-R158),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL292-R284),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL431-R385),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL641-R595),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL721-R675),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL835-R790),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL867-R835),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-4cf121312b4d1903ac9893f9cfd398842158689001fdd79351a5cf74194c4f2cL933-R841))
*  Updated the `useOrgShareToken` hook to accept a `type` parameter of the `ShareTokenType` enum, which is used to distinguish between invite and welcome tokens for an organization. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-237630bdfb8e4278c7faf289bbeaabb329a45e5ef7d8c8a5edcfe1e1355134f7L8-R10),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-a4786faf76b2e5091663392a05b3bbe28e71dba5f67925454e12ac413cc50ff7L5-R9),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-a4786faf76b2e5091663392a05b3bbe28e71dba5f67925454e12ac413cc50ff7L21-R41))
*  Updated the `AddMembersPage` component to use a conditional expression based on the `groupType` prop to determine whether the link should point to the circle or the organization settings page with the `#guild` hash. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-55b839895ad7ab38d72feef621e328ad31f586d502d7d5561293b642c65b8032L22-R22),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-55b839895ad7ab38d72feef621e328ad31f586d502d7d5561293b642c65b8032L293-R297),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-55b839895ad7ab38d72feef621e328ad31f586d502d7d5561293b642c65b8032L330-R338))
*  Updated the `AddPage` component to add the `welcomeLink` prop to the `AddMembersContents` component, which is used to display the welcome link for an organization. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-a4786faf76b2e5091663392a05b3bbe28e71dba5f67925454e12ac413cc50ff7L81-R95))
*  Updated the `AddressIsNotMember` component to use a conditional expression based on the `tokenJoinInfo.circle` value to determine whether the user is joining a circle or an organization, and to display the guild requirement and the redirect link accordingly. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L9-R10),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L35-R49),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L55-R61),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L66-R77),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L87-R87),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-398496b2c2b4680fd1412adbec64e025fb287e999f27ebfea7619482dfd6c086L118-R122))
*  Updated the `getOrgData` function to include the `guild_id` and `guild_role_id` fields in the query for the `organizations` table, which are used to determine the guild membership requirement for joining an organization. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-2cad621541bf831d327fcdf87e2186c3241920cf35a3eb92b06236e8d94e254aR14-R15))
*  Updated the `useFetchManifest` hook to fetch the manifest data from the legacy API and update the recoil state. ([link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-e62b950b94784333ac920a0cdd946b27d7a7c292af43c1cfcc541ece0fa312e1L9-R10),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-e62b950b94784333ac920a0cdd946b27d7a7c292af43c1cfcc541ece0fa312e1R101-R102),[link](https://github.com/coordinape/coordinape/pull/2126/files?diff=unified&w=0#diff-bfd2a0da57dcf081d9374ed741f4713e5873c08141a2d4a10f27b976dde66e4fL57-R74))

### CODE REVIEW

File: src/features/guild/GuildSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  guildInfo,
+  formControl,
+  register,
+  guild_role_id,
+  setValue,
+  setGuildInfo,
+  guild_id,
+  isOrg = false,
+}: {
+  formControl: Control<any>; // TODO: this any is troubling, I couldn't get generics to work properly
```
Comment (CryptoGraffe) [inline src/features/guild/GuildSelector.tsx:28]: I'd like to improve this in future work - as well as add a test w/ @levity help

File: src/features/orgs/OrgSettingsPage.tsx [hunk 2]
```diff
@@ -75,8 +92,16 @@ export const OrgSettingsPage = () => {
...
-      await updateOrg(orgId, data);
+      const d = {
+        ...data,
+        // these need to be null to clear them
+        guild_id: guildInfo?.id ?? null,
+        guild_role_id: data.guild_role_id ?? null,
+      };
+      await updateOrg(orgId, d);
       refetch();
+      await fetchManifest();
```
Comment (CryptoGraffe) [inline src/features/orgs/OrgSettingsPage.tsx:103]: @levity this is an attempt to get the useLoginData stuff to reload for the `AddPage` but it isn't working - need some help here.

File: api/hasura/actions/_handlers/createUserWithToken.ts [hunk 3]
```diff
@@ -141,6 +145,20 @@ const handleOrg = async (
 
   const org = org_share_tokens[0]?.organization;
   if (!org) return false;
+  if (org_share_tokens[0]?.type != ShareTokenType.Invite) {
+    // if this is just a welcome link, we need to see if they are a guild member, otherwise they can't join
```
Comment (crabsinger) [inline api/hasura/actions/_handlers/createUserWithToken.ts:149]: `  if (org_share_tokens[0]?.type != ShareTokenType.Invite)` if the share type is not an invite, does that mean its a welcome link?

I would expect this to be `if share_token_type == welcome link` not is not Invite... ?

**Review by crabsinger — APPROVED** [review]
lgtm

**Pull request merged**